src/pgp/gpg.c

```diff
@@ -303,67 +307,62 @@ p_gpg_free_key(ProfPGPKey* key)
     }
 }
 
+/**
+ * Retrieve a list of GPG keys and create a hash table of ProfPGPKey objects.
+ *
+ * This function utilizes the GPGME library to retrieve both public and secret keys.
+ * It iterates over the keys and their subkeys to populate a hash table with ProfPGPKey objects.
+ * The key name is used as the key in the hash table, and the ProfPGPKey object is the corresponding value.
+ *
+ * @return A newly created GHashTable* containing ProfPGPKey objects, with the key name as the key.
+ *         Returns NULL if an error occurs during key retrieval or if memory allocation fails.
+ *
+ * @note The returned hash table should be released using p_gpg_free_keys()
+ *       when they are no longer needed to avoid memory leaks.
+ *
+ * @note This function may perform additional operations, such as autocomplete, related to the retrieved keys.
+ */
 GHashTable*
 p_gpg_list_keys(void)
 {
     gpgme_error_t error;
     GHashTable* result = g_hash_table_new_full(g_str_hash, g_str_equal, free, (GDestroyNotify)p_gpg_free_key);
 
+    // Create a GPGME context
```

Comment:
https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/

> `Rule 1: Comments should not duplicate the code.`

This applies to all comments in the code you added in this PR that I've found so far.